### ISSUE TITLE
 No Trophy Given After Purchasing Compendium 2024 

### ISSUE THREAD

**Issue by cbq1232006** [ValveSoftware/Dota2-Gameplay#21142]
### Description

As per the title, I bought the Compendium, but I didn't receive the trophy or any trophy levels.

A Level 1 trophy should have been granted to me. I checked with all my friends who bought it, and none of us received the trophy or levels. 

![QQ截图20240823174154](https://github.com/user-attachments/assets/14c3b81e-108d-43a3-88a9-1f03cf8440de)
![QQ截图20240823174234](https://github.com/user-attachments/assets/f3f90935-e1e3-4a9d-9e40-44458adade4d)
![QQ截图20240823174644](https://github.com/user-attachments/assets/0a4651d5-6ed1-42b6-8b53-5b2bc7a0adc4)

**Comment by MrRobotDota** [comment]
To add to this: I didn't have a compendium in 2023, but have now received the trophy for the compendium 2023 with the purchase of the compendium 2024 and the levels are also increasing here. Trophy compendium 2023 is now level 52, but trophy compendium 2024 is level 0

**Comment by cbq1232006** [comment]
> To add to this: I didn't have a compendium in 2023, but have now received the trophy for the compendium 2023 with the purchase of the compendium 2024 and the levels are also increasing here. Trophy compendium 2023 is now level 52, but trophy compendium 2024 is level 0

Yes, I checked one of my friend's trophy. Same issue appeared, trophy level of compendium 2024 is assigned to compendium 2023 trophy. 

![QQ截图20240824023234](https://github.com/user-attachments/assets/093f34fd-5d39-4fae-b07a-18841d091c7b)
![QQ截图20240824023219](https://github.com/user-attachments/assets/4972931d-7567-4be1-b413-25d577a2ce3e)
